opentelemetry-api/src/opentelemetry/resources/__init__.py

```diff
@@ -11,3 +11,56 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+from typing import Any, Dict, Union
+
+
+class Resource:
+    def __init__(self, labels: Dict[str, str]):
+        """
+        Construct a resource. Direct calling of the
+        constructor is discouraged, as it cannot
+        take advantage of caching and restricts
+        to the type of object that can be returned.
+        """
+        self._labels = labels
+
+    @staticmethod
+    def create(labels: Dict[str, str]) -> "Resource":
+        """
+        create a new resource. This is the recommended
+        method to use to create a new resource.
+        """
+        return Resource(labels)
+
+    def merge(self, other: Union["Resource", None]) -> "Resource":
+        """
+        Perform a merge of the resources, resulting
+        in a union of the resource objects.
+
+        labels that exist in the main Resource take
+        precedence unless the label value is empty.
+        """
+        if other is None:
+            return self
+        if not self._labels:
+            return other
+        merged_labels = self.get_labels().copy()
```

Comment:
We probably want to move these implementations from API to SDK package.
The spirit here is that we want API package to be the minimum thing to keep the app working.

Reply:
yes, will move to SDK